flink-streaming-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -318,6 +312,46 @@ protected void verifyResultsDataDiscardingUponRestore(
 		Assert.assertTrue("The following ID's were not found in the ResultSet: " + list.toString(), list.isEmpty());
 	}
 
+	@Override
+	protected void verifyResultsWhenScalingDown(CassandraTupleWriteAheadSink<Tuple3<String, Integer, Integer>> sink) throws Exception {
+
+		ArrayList<Integer> list = new ArrayList<>();
+		for (int x = 1; x < 34; x++) {
+			list.add(x);
+		}
+
+		ResultSet result = session.execute(SELECT_DATA_QUERY);
+		for (Row s : result) {
+			list.remove(new Integer(s.getInt("counter")));
```

Comment:
you should check whether remove returns false; this would imply that we committed to many values.